portal/src/main/webapp/js/src/oncoprint/webgl/setup.js

```diff
@@ -791,6 +873,39 @@ window.CreateCBioPortalOncoprintWithToolbar = function (ctr_selector, toolbar_se
 		oncoprint.releaseRendering();
 		return track_ids;
 	    },
+	    'addHeatmapTracks': function (heatmap_data_by_line) {
+		oncoprint.suppressRendering();
+		var hm_ids = [];
+		for (var i = 0; i < heatmap_data_by_line.length; i++) {
+		    var track_params = {
+			'rule_set_params': {
+			    'type': 'gradient',
+			    'value_key': 'profile_data',
+			    'value_range': [-3, 3],
+			    'colormap': 'inferno',
+			    //'color_range': ['rgba(0,0,255,1)', 'rgba(255,0,0,1)'],
+			    'null_color': 'rgba(211,211,211,1)'
+			},
+			//'track_padding': 0,
+			'label': heatmap_data_by_line[i].hugo_gene_symbol,
+			'target_group': 2,
+			//'sortCmpFn': function(d1, d2) {return 0;},
+			'removable': true,
+			'description': heatmap_data_by_line[i].hugo_gene_symbol,
+		    };
+		    var new_hm_id = oncoprint.addTracks([track_params])[0];
+		    hm_ids.push(new_hm_id);
+		    State.heatmap_tracks[i] = new_hm_id;
+		    // ???: wouldn't this always be State.heatmap_track[0]? <fedde-s 2016-10-06>
```

Comment:
Remove comment, but I agree we don't need `State.first_heatmap_track` - replace the `if` test with a check for the length of heatmap_tracks, and replace `State.first_heatmap_track` with `State.heatmap_tracks[0]`